test/integration/tests/tribalchief.ts

```diff
@@ -0,0 +1,75 @@
+import chai, { expect } from 'chai';
+import CBN from 'chai-bn';
+import { solidity } from 'ethereum-waffle';
+import { ethers } from 'hardhat';
+import { NamedContracts } from '@custom-types/types';
+import proposals from '@test/integration/proposals_config';
+import { TestEndtoEndCoordinator } from '@test/integration/setup';
+import mainnetAddresses from '@protocol/mainnetAddresses';
+import tribalchief from '@protocol/tribalchief';
+
+describe('e2e-tribalchief', function () {
+  let contracts: NamedContracts;
+  let e2eCoord: TestEndtoEndCoordinator;
+  const doLogging = Boolean(process.env.LOGGING);
+
+  before(async () => {
+    chai.use(CBN(ethers.BigNumber));
+    chai.use(solidity);
+  });
+
+  before(async function () {
+    const config = {
+      logging: doLogging,
+      deployAddress: (await ethers.getSigners())[0].address,
+      version: 1
+    };
+    e2eCoord = new TestEndtoEndCoordinator(config, proposals);
+    ({ contracts } = await e2eCoord.loadEnvironment());
+  });
+
+  it('should have the proper total number of pools', async function () {
+    const numPools = await contracts.tribalChief.numPools();
+    expect(numPools).to.be.equal(Object.keys(tribalchief).length);
+  });
+
+  it('should have the proper total of allocated points', async function () {
+    const totalAllocPoint = await contracts.tribalChief.totalAllocPoint();
+    const totalAllocPointDocumented = Object.values(tribalchief).reduce((sum, cur) => {
+      sum += cur.allocPoint;
+      return sum;
+    }, 0);
+    expect(totalAllocPoint).to.be.equal(totalAllocPointDocumented);
+  });
+
+  it('should properly describe the current pools config', async function () {
+    const numPools = await contracts.tribalChief.numPools();
+    const documentedPoolsKeys = Object.keys(tribalchief);
+    const documentedPoolsValues = Object.values(tribalchief);
+    for (let i = 0; i < numPools && i < documentedPoolsKeys.length; i++) {
+      const documentedStakedContract = mainnetAddresses[documentedPoolsKeys[i]];
+      const errmsg1 = `Staked token "${documentedPoolsKeys[i]}" is not found in mainnetAddresses.ts`;
```

Comment:
is this the actual error when you dereference an object that doesn't exist?

Reply:
I would think the error message would be something like 'property x is undefined'